toqito/perms/permute_systems.py

```diff
@@ -210,7 +210,7 @@ def permute_systems(
     row_perm = permute_systems(vec_arg, perm, dim[0][:], False, inv_perm)
 
     # This condition is only necessary if the `input_mat` variable is sparse.
-    if isinstance(input_mat, (sparse.csr_matrix, sparse.dia_matrix)):
+    if isinstance(input_mat, (sparse.csr_array, sparse.dia_array)):
```

Comment:
This change is obviously not working as somewhere in the code `input_mat` is a `sparse.dia_matrix`. This causes the code to move to the `else` block i.e. tests fail. Need to figure out which line is creating the `sparse.dia_matrix`.